appsdk/functions_test.go

```diff
@@ -0,0 +1,154 @@
+		assert.Equal(t, path, r.URL.EscapedPath(), "Invalid path received %s, expected %s", r.URL.EscapedPath(), path)
+	}
+	pushedHandler := func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		assert.Contains(t, r.URL.EscapedPath(), "234")
+	}
+	// create test server with handler
+	ts := httptest.NewServer(http.HandlerFunc(handler))
+	ts2 := httptest.NewServer(http.HandlerFunc(pushedHandler))
+	defer ts.Close()
+	defer ts2.Close()
+	//Setup eventClient
+	params := types.EndpointParams{
+		ServiceKey:  clients.CoreDataServiceKey,
+		Path:        clients.ApiEventRoute,
+		UseRegistry: false,
+		Url:         ts2.URL,
+		Interval:    1000,
+	}
+	eventClient := coredata.NewEventClient(params, startup.Endpoint{RegistryClient: nil})
+
+	context = &appcontext.Context{
+		EventID:       "234",
+		LoggingClient: lc,
+		EventClient:   eventClient,
+	}
+	// used to ensure MarkAsPushed is called
+	trx := sdk.HTTPPostJSON(ts.URL)
+	assert.NotNil(t, trx, "return result from HTTPPostJSON should not be nil")
+
+	result, data := trx(context, msgStr)
+	assert.True(t, result, "continuePipeline should be true")
+	assert.Equal(t, "RESPONSE", (string)((data).([]byte)), "response should be \"RESPONSE\"")
+}
+func TestHTTPPostXML(t *testing.T) {
+	sdk := AppFunctionsSDK{
+		LoggingClient: lc,
+	}
+	msgStr := "POST ME"
+	path := "/"
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write(([]byte)("RESPONSE"))
+		readMsg, _ := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		assert.Equal(t, msgStr, (string)(readMsg), "Invalid msg received %v, expected %v", (string)(readMsg), msgStr)
+		assert.Equal(t, "application/xml", r.Header.Get("Content-type"), "Unexpected content-type received %s, expected %s", r.Header.Get("Content-type"), "application/xml")
+		assert.Equal(t, path, r.URL.EscapedPath(), "Invalid path received %s, expected %s", r.URL.EscapedPath(), path)
+	}
+	// used to ensure MarkAsPushed is called
+	pushedHandler := func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		assert.Contains(t, r.URL.EscapedPath(), "123")
+	}
+	// create test server with handler
+	ts := httptest.NewServer(http.HandlerFunc(handler))
+	ts2 := httptest.NewServer(http.HandlerFunc(pushedHandler))
+	defer ts.Close()
+	defer ts2.Close()
+	//Setup eventClient
+	params := types.EndpointParams{
+		ServiceKey:  clients.CoreDataServiceKey,
+		Path:        clients.ApiEventRoute,
+		UseRegistry: false,
+		Url:         ts2.URL,
+		Interval:    1000,
+	}
+	eventClient := coredata.NewEventClient(params, startup.Endpoint{RegistryClient: nil})
+
+	context = &appcontext.Context{
+		EventID:       "123",
+		LoggingClient: lc,
+		EventClient:   eventClient,
+	}
+
+	trx := sdk.HTTPPostXML(ts.URL)
+	assert.NotNil(t, trx, "return result from HTTPPostXML should not be nil")
+
+	result, data := trx(context, msgStr)
```

Comment:
Same above already tested in the transform unit test.

Reply:
doesn't look like it, if i remove the test, it shows as not being tested